firmware/quadruna/VC/src/app/states/app_faultState.c

```diff
@@ -0,0 +1,79 @@
+#include "app_faultState.h"
+#include "app_initState.h"
+#include "app_allStates.h"
+
+#include "app_canUtils.h"
+#include "app_canTx.h"
+#include "app_canRx.h"
+
+#include "app_powerManager.h"
+#include "app_faultCheck.h"
+
+// TODO: When should we be entering fault state?
+
+static const PowerStateConfig power_manager_fault = {
+    .efuses = {
+        [EFUSE_CHANNEL_SHDN] = true,
+        [EFUSE_CHANNEL_LV] = true,
+        [EFUSE_CHANNEL_PUMP] = false,
+        [EFUSE_CHANNEL_AUX] = false,
+        [EFUSE_CHANNEL_INV_R] = false,
+        [EFUSE_CHANNEL_INV_L] = false,
+        [EFUSE_CHANNEL_TELEM] = true,
+        [EFUSE_CHANNEL_BUZZER] = false,
+    },
+};
+
+static void faultStateRunOnEntry(void)
+{
+    app_canTx_VC_State_set(VC_FAULT_STATE);
+    app_powerManager_updateConfig(power_manager_fault);
+    app_canTx_VC_Fault_StateMachine_set(true);
+
+    // Disable inverters and apply zero torque upon entering Fault state
+    app_canTx_VC_LeftInverterEnable_set(false);
+    app_canTx_VC_RightInverterEnable_set(false);
+    app_canTx_VC_LeftInverterTorqueCommand_set(0.0f);
+    app_canTx_VC_RightInverterTorqueCommand_set(0.0f);
+    app_canTx_VC_LeftInverterTorqueLimit_set(0.0f);
+    app_canTx_VC_RightInverterTorqueLimit_set(0.0f);
+    app_canTx_VC_LeftInverterDirectionCommand_set(INVERTER_REVERSE_DIRECTION);
+    app_canTx_VC_RightInverterDirectionCommand_set(INVERTER_FORWARD_DIRECTION);
+}
+
+static void faultStateRunOnTick1Hz(void)
+{
+    app_allStates_runOnTick1Hz();
+}
+
+static void faultStateRunOnTick100Hz(void)
```

Comment:
We should be running app_allStates_runOnTick100Hz in every states 100Hz task since this is where we are broadcasting the information